### ISSUE TITLE
running time

### ISSUE THREAD

**Issue by YunYang1994** [pytorch/pytorch#13115]
I detach [ATen](https://github.com/zdevito/ATen/tree/master/aten) library from pytorch and compile them into dynamic library (".so" files), then target my neural network model(".cpp" files) into the ATen (".so") and compile it. *My question is that It seems running on ATen library need much more time than pytorch(GPU). I don't think this is reasonable !*
## running time
| iterations | 10 | 100 | 1000 | 5000 |
| ---------- | -----------| ---------- | -----------| ---------- |
| `pytorch(GPU)`   | 0.1174318790 s  | 1.107207059 s | 11.13777494 s | 55.82848405 s |
| `pytorch(CPU)`   | 0.7973520755 s  | 7.873554949 s | 78.74021005 s |  392.61105028 s |
| `ATen`   | 0.805274 s  | 1.8661 s | 11.983 s | 58.0243 s |

here is my code for both pytorch and ATen to test resnet18 model
## using pytorch
```py
from src.resnet18_onnx import Predictor
import torch
import time

model = Predictor()
image = torch.ones(1,3,224,224).to('cpu')
# model.to('cpu')
# result = model.predict(image)

start = time.time()
iteration =1000
for i in range(iteration):
    result = model.predict(image)
result.detach()
end = time.time()
print end - start
```

## using ATen library
```cpp
#include <iostream>
#include <memory>
#include <vector>
#include <ATen/ATen.h>
#include <time.h>
// #include <ATen/Context.h>
#include <predict.h>  // see predict.h in the include directory
using namespace at;

int main(int argc, char** argv) {

    std::shared_ptr<Predictor> a = std::make_shared<Predictor> ();
    std::vector<float> data(1*3*224*224, 1);

    int iteration = 1000;
    clock_t t_start, t_end;
    double totaltime;

    t_start = clock();
    for(int i=0; i<iteration; i++){
        // std::cout << a->predict(&data[0]) << std::endl;
        a->predict(&data[0]);
    }
    t_end = clock();
    totaltime = (double)(t_end - t_start)/CLOCKS_PER_SEC;
    std::cout << totaltime << std::endl;

}
```
[predict.h](https://github.com/YunYang1994/Onnx2Aten/blob/master/include/predict.h) 

more details please see [my code](https://github.com/YunYang1994/Onnx2Aten), do you have any ideas about it? Looking forward to your reply!

**Comment by YunYang1994** [comment]
@svenstaro @eklitzke @huitseeker @dreiss hello, can you help me ?

**Comment by ezyang** [comment]
Your ATen code is written to run the model on CPU, and in fact the performance matches PyTorch CPU. You need to transfer inputs to CUDA if you want to use GPU acceleration.
